XPCBackupService/Services/BackupRealm.swift

```diff
@@ -8,14 +8,20 @@
 import Foundation
 import RealmSwift
 
-protocol BackupRealmProtocol {
-    func getRealm() throws -> Realm?
-    func addSyncedNode(_ node: SyncedNode) throws
+//protocol BackupRealmProtocol {
+//    func getRealm() throws -> Realm?
+//    func addSyncedNode(_ node: SyncedNode) throws
+//    func findSyncedNode(url: URL, deviceId: Int) -> ThreadSafeReference<SyncedNode>?
+//    func editSyncedNodeDate(remoteUuid: String, date: Date) throws
+//}
+
+protocol BackupRealmProtocol: GenericRepositoryProtocol where T == SyncedNode {
     func findSyncedNode(url: URL, deviceId: Int) -> ThreadSafeReference<SyncedNode>?
-    func editSyncedNodeDate(remoteUuid: String, date: Date) throws
+    func getRealm() throws -> Realm?
 }
 
 struct BackupRealm : BackupRealmProtocol {
```

Comment:
Try to abstract `Realm` itself, the consumer layer should not be aware of the dependency we are using behind, this will make difficult to remove `Realm` in the future if we want to